### PR TITLE
Add type check trouble shooting to FAQ

### PR DESCRIPTION
**Pull request by kmaehashi**
It is a common pitfall esp. for beginners. (#4783)

### CODE REVIEW

File: docs/source/tips.rst [hunk 1]
```diff
@@ -48,6 +48,58 @@ If you want to install NumPy with pip, use `site.cfg <https://github.com/numpy/n
...
+   Traceback (most recent call last):
+   ...
+   chainer.utils.type_check.InvalidType:
+   Invalid operation is performed in: ReLU (Forward)
+
+   Expect: in_types[0].dtype.kind == f
+   Actual: i != f
+
+In this case, :class:`~numpy.dtype.kind` of ``in_types[0]`` (which means the first input to the function, ``x``) is expected to be ``f`` (floating-point), whereas the input was ``i`` (signed integer).
+You need to cast the input appropriately before passing to the function (e.g., ``x.asarray(np.float32)``).
```
Comment (beam2d) [inline docs/source/tips.rst]: s/asarray/astype/

Comment (kmaehashi) [reply docs/source/tips.rst]: Thanks, fixed!

**Comment by kmaehashi** [comment]
@asi1024 Could you takeover the assignee? (review is almost complete as above)

**Comment by asi1024** [comment]
OK,

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**